adapters/iqzone/iqzone.go

```diff
@@ -23,19 +23,70 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters
 	return bidder, nil
 }
 
-func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	requestJSON, err := json.Marshal(request)
-	if err != nil {
-		return nil, []error{err}
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var err error
+	var bidderExt adapters.ExtImpBidder
+	var iqzoneExt openrtb_ext.ImpExtIQZone
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for i, imp := range request.Imp {
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		if err = json.Unmarshal(reqCopy.Imp[i].Ext, &bidderExt); err != nil {
```

Comment:
should this be `json.Unmarshal(reqCopy.Imp[0].Ext, &bidderExt)`? Line 35 above you have defined `reqCopy.Imp[]` to have only one entry in its array.

Reply:
oh, thanks for finding this problem, we fixed it